modal/_serialization.py

```diff
@@ -42,13 +46,142 @@ def persistent_load(self, pid):
             raise InvalidError("bad flag")
 
 
-def serialize(obj):
+def serialize(obj: Any) -> bytes:
     """Serializes object and replaces all references to the client class by a placeholder."""
     buf = io.BytesIO()
     Pickler(buf).dump(obj)
     return buf.getvalue()
 
 
-def deserialize(s: bytes, client):
+def deserialize(s: bytes, client) -> Any:
     """Deserializes object and replaces all client placeholders by self."""
     return Unpickler(client, io.BytesIO(s)).load()
+
+
+def _serialize_asgi(obj: Any) -> api_pb2.Asgi:
```

Comment:
The changes in this file add code to serialize and deserialize ASGI messages. This makes an assumption that they are well-formed. It can throw an exception if the `obj` is an invalid ASGI message, but at that point, it's a bug with whatever web framework the user is using.